docs/src/style.md

```diff
@@ -1,15 +1,78 @@
 Style guide and design principles
 =================================
 
+This document outlines JuMP's style guide. It also contains a set or principles
+for users writing JuMP models.
+
 Style guide
 -----------
 
-TODO: A style guide for JuMP, JuMP models and surrounding Julia code.
-Formatting, naming, use of macros, comments, TODOs, docstrings, etc.
+!!! info
+    For developers: if you modify JuMP's source code, please fix the style of
+    the surrounding code while you are there.
+
+### JuMP Source Code
+
+ - 80 character line limit. (Except for long URLs in comments.)
```

Comment:
I'm REALLY not a fan of character line limits. most modern code editors automatically wrap at an appropriate length given the specific user's display size. I think line limits paradoxically make code harder to read, and they make it more work to update (eg if you change a function name or a comment you often have to reformat the remaining lines to even the lengths out).

Reply:
I do however, like [semantic linebreaking](http://rhodesmill.org/brandon/2012/one-sentence-per-line/). it is particularly convenient with git diffs too.

could we recommend semantic linebreaking instead of a hard 80 char rule?